lib/jekyll/document.rb

```diff
@@ -262,8 +284,12 @@ def published?
     def read(opts = {})
       Jekyll.logger.debug "Reading:", relative_path
 
-      if yaml_file?
+      if yaml_file? || json_file?
         @data = SafeYAML.load_file(path)
+      elsif csv_file?
+        @data = CSV.read(path,
```

Comment:
We can check for .tsv here and use "," for .csv and "\t" for .tsv as the col_sep so we dont have to make two different paths. 

```ruby
:col_sep  => column_delimiter, 
```

and the method:
```ruby


    # Determine the column delimiter for ___SeparatedValue files.
    #
    # Returns "\t" if extname is `.tsv`, returns "," otherwise 
    def column_delimiter
      return "\t" if extname == ".tsv"
      ","
    end
```

Reply:
this also allows for future updates if separator is bar or colon separated...